### PR TITLE
Cli save config

### PR DESCRIPTION
**Pull request by xujianin1988**
Check in for issue #94 #93 #42 #10 
Summary: 
1. Add command config to show/list/save/load/delete configuration.
2. Add restriction for adding/deleting many-to-many relationship.
3. Enable deletion for non-restrict model include it's relationships.
4. Fix bug that hit enter will execute previous command.
5. Implement delete_all in CPDKd.

### CODE REVIEW

**Comment by xujianin1988** [comment]
Need to create another issue for the documentation of config command.

**Comment by xujianin1988** [comment]
Sorry, I also added all the unit tests related to new features in this check-in.

File: examples/basic/models/model.py [hunk 1]
```diff
@@ -35,7 +35,8 @@ class Server(CPDKModel):
                            'negative_cmd_help': 'administratively disable the server',
                            'required_fields': ['address', 'port']})
     virtual_servers = relationship('VirtualServer',
-                                   secondary=Server_VS_Map)
+                                   secondary=Server_VS_Map,
+                                   info={"restrict": True})
```
Comment (zimventures) [inline examples/basic/models/model.py]: Add a note describing which side of the relationship this is restricting

File: redshell.py [hunk 2]
```diff
@@ -565,17 +672,290 @@ def help_show(self):
...
+    base_def += '''
+                    f_config = open(file_path, "r")
+                    file_content = f_config.read()
+                    f_config.close()
+                    cmd_list = file_content.split("\\n")
+                    cmd_list = filter(None, cmd_list)
+                    current_model = ""
+                    current_model_name = ""
+                    for cmd in cmd_list:
+                        print "Processing cmd: %s" % cmd
```
Comment (zimventures) [inline redshell.py]: I don't think you want to actually print this stuff out to the console. For debugging, sure, but not in production.

File: redshell.py [hunk 3]
```diff
@@ -565,17 +672,290 @@ def help_show(self):
...
+            if len(arg_list) < 2:
+                print "*** ERROR: Not enough arguments (config load <file name>)"
+                return
+            else:
+                file_name = arg_list[1]
+                print "Loading %s" % file_name
+                file_path = os.path.join(self.config_path, file_name)
+                pass_flg = True
+                if not os.path.isfile(file_path):
+                    print "Configuration file does not exist."
```
Comment (zimventures) [inline redshell.py]: For errors, print "*** ERROR: <your message"

**Comment by xujianin1988** [comment]
Thanks man ! I will address these issues.

**Pull request merged**